### PR TITLE
[Macros] Track macro dependency separately in module trace

### PR DESCRIPTION
**Pull request by rintaro**
Macro plugins are not normal Swift modules, track them differently. Add "swiftmacros" field to the JSON file.

rdar://110927706

### CODE REVIEW

**Review by owenv — APPROVED** [review]

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: lib/AST/PluginLoader.cpp [hunk 1]
```diff
@@ -213,7 +209,25 @@ PluginLoader::loadExecutablePlugin(StringRef path) {
...
+  // value.
+
+  assert(!path.empty());
+  SmallString<128> resolvedPath;
+  auto fs = Ctx.SourceMgr.getFileSystem();
+  if (auto err = fs->getRealPath(path, resolvedPath)) {
+    return;
+  }
+
+  DepTracker->addMacroPluginDependency(path, moduleName);
```
Comment (DougGregor) [inline lib/AST/PluginLoader.cpp:232]: Did you mean to pass `resolvedPath` here?

File: lib/FrontendTool/MakeStyleDependencies.cpp [hunk 2]
```diff
@@ -132,6 +132,12 @@ bool swift::emitMakeDependenciesIfNeeded(DiagnosticEngine &diags,
     dependencyString.push_back(' ');
     dependencyString.append(frontend::utils::escapeForMake(path, buffer).str());
   }
+  auto macroPluginDependencyPath =
+      reversePathSortedFilenames(depTracker->getIncrementalDependencyPaths());
+  for (auto const &path : macroPluginDependencyPath) {
+    dependencyString.push_back(' ');
+    dependencyString.append(frontend::utils::escapeForMake(path, buffer).str());
+  }
```
Comment (DougGregor) [inline lib/FrontendTool/MakeStyleDependencies.cpp:140]: Good catch. I don't see a test for this, though?

Comment (rintaro) [reply lib/FrontendTool/MakeStyleDependencies.cpp:140]: Actually, this has an error `getIncrementalDependencyPaths()` -> `getMacroPluginDependencyPaths()`. Thanks for pointing it out!

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Pull request merged**